### PR TITLE
Manual analysis 301-350

### PR DESCRIPTION
**Pull request by LHolten**
For some commits i filled in a question mark for the Category.
After fixing those this pull request will be ready.

### CODE REVIEW

**Comment by LHolten** [comment]
Ready for review

File: analysis/PX4-Autopilot.md [hunk 1]
```diff
@@ -5201,11 +5204,11 @@ runtime
 ### Message
 nuttx defconfig: increase nr of file descriptors, due to mavlink shell
 ### Antipattern Category
-
+New:Hard-coded-fine-tuning
 ### Keyword
 increase
 ### Note
-
+The number of file descriptors is changed without explanation.
```
Comment (pderakhshanfar) [inline analysis/PX4-Autopilot.md:5211]: But, do we know if these changes are because of a performance issue?

File: analysis/PX4-Autopilot.md [hunk 2]
```diff
@@ -5338,11 +5346,11 @@ memory
 ### Message
 Make altitude more efficient and estimator status safe in terms of memory overflow
 ### Antipattern Category
-
+X
 ### Keyword
 memory
 ### Note
-
+No anti-pattern was found.
```
Comment (pderakhshanfar) [inline analysis/PX4-Autopilot.md:5353]: Can you elaborate more about this case?

File: analysis/PX4-Autopilot.md [hunk 3]
```diff
@@ -5557,11 +5566,12 @@ is called from the work queue thread. Sending the signal creates measurable
 overhead (~5% of the px4 CPU runtime) and is unnecessary, since the thread
 is not sleeping anyway.
 ### Antipattern Category
-
+X
 ### Keyword
 runtime
 ### Note
-
+There is no need to wake up the worker thread if that thread is the current thread.
```
Comment (pderakhshanfar) [inline analysis/PX4-Autopilot.md:5573]: Isnt it "Smith:General:Unnecessary_Processing"?

**Review by pderakhshanfar — CHANGES_REQUESTED** [review]

File: analysis/PX4-Autopilot.md [hunk 4]
```diff
@@ -4805,11 +4806,11 @@ increase
 ### Message
 Navigator: Run faster
 ### Antipattern Category
-
+X
 ### Keyword
 faster
 ### Note
-
+No antipatterns were found.
```
[the remark this answers is not in the document]
Comment (LHolten) [reply analysis/PX4-Autopilot.md:4813]: You may be right, it's hard to tell.

File: analysis/PX4-Autopilot.md (hunk 1 above)
Comment (LHolten) [reply analysis/PX4-Autopilot.md:5211]: There are no issues related to this commit. Searching for file descriptors turned up this https://github.com/PX4/PX4-Autopilot/issues/14157#issuecomment-588372647. So the commit under analysis was probably also to fix a similar problem.
This leads me to believe it is not a performance anti pattern.

Comment (LHolten) [reply analysis/PX4-Autopilot.md:5211]: Why would they choose a small number of file descriptors if not for performance?

File: analysis/PX4-Autopilot.md (hunk 2 above)
Comment (LHolten) [reply analysis/PX4-Autopilot.md]: Done, you can check again.

File: analysis/PX4-Autopilot.md (hunk 3 above)
Comment (LHolten) [reply analysis/PX4-Autopilot.md:5576]: Fixed

File: analysis/PX4-Autopilot.md [hunk 5]
```diff
@@ -5247,11 +5252,11 @@ We should not spam the console just because the input data is
 degenerate, it would only make things worse because everything would
 slow down due to the printfs.
 ### Antipattern Category
-
+General:Unnecessary-processing
```
[the remark this answers is not in the document]
Comment (LHolten) [reply analysis/PX4-Autopilot.md]: Fixed

File: analysis/PX4-Autopilot.md [hunk 6]
```diff
@@ -4896,11 +4898,11 @@ clang failed with:
 size of 7400 bytes in function 'test_mathlib' [-Wframe-larger-than=]
 int test_mathlib(int argc, char *argv[])
 ### Antipattern Category
-
+New:Hard-coded-fine-tuning
 ### Keyword
 increase
 ### Note
-
+Changes the stack size of some tests.
```
[the remark this answers is not in the document]
Comment (LHolten) [reply analysis/PX4-Autopilot.md:4906]: Fixed

File: analysis/PX4-Autopilot.md (hunk 1 above)
Comment (pderakhshanfar) [reply analysis/PX4-Autopilot.md:5212]: I see your point here.
can you extend the note and explain that the authors had some discussions about the limited number of file descriptors? and also add the link to the comment

**Pull request merged**